### PR TITLE
[POAE7-2759] Migrate substrait-cpp's quotation from Intel-bigdata to substrait-io

### PR DESCRIPTION
**Pull request by haiweizh**
### What changes were proposed in this pull request?

Migrate substrait-cpp's quotation from Intel-bigdata to substrait-io

### Why are the changes needed?

Migrate substrait-cpp's quotation from Intel-bigdata to substrait-io

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Uts.

### Which label does this PR belong to?

cider

### CODE REVIEW

File: cpp/src/cider/function/FunctionLookupEngine.cpp [hunk 1]
```diff
@@ -375,10 +375,193 @@ const FunctionDescriptor FunctionLookupEngine::lookupFunction(
...
+        return std::make_shared<io::substrait::Varchar>(0, nullable);
+      }
+      return std::make_shared<io::substrait::Varchar>(std::stoi(length->value()),
+                                                      nullable);
+    } else if (io::substrait::TypeTraits<
+                   io::substrait::TypeKind::kFixedChar>::typeString == baseType) {
+      auto length =
+          std::dynamic_pointer_cast<const io::substrait::StringLiteral>(nestedTypes[0]);
+      if (!length->value().empty() && length->value().at(0) == 'L') {
+        return std::make_shared<io::substrait::Varchar>(0, nullable);
```
Comment (chaojun-zhang) [inline cpp/src/cider/function/FunctionLookupEngine.cpp:543]: this should be FixedVar

File: cpp/src/cider/function/FunctionLookupEngine.cpp [hunk 2]
```diff
@@ -375,10 +375,193 @@ const FunctionDescriptor FunctionLookupEngine::lookupFunction(
...
+      }
+      return std::make_shared<io::substrait::FixedChar>(std::stoi(length->value()),
+                                                        nullable);
+    } else if (io::substrait::TypeTraits<
+                   io::substrait::TypeKind::kFixedBinary>::typeString == baseType) {
+      auto length =
+          std::dynamic_pointer_cast<const io::substrait::StringLiteral>(nestedTypes[0]);
+      if (!length->value().empty() && length->value().at(0) == 'L') {
+        return std::make_shared<io::substrait::Varchar>(0, nullable);
```
Comment (chaojun-zhang) [inline cpp/src/cider/function/FunctionLookupEngine.cpp:552]: this should be FixedBinary

File: cpp/src/cider/function/FunctionLookupEngine.cpp (hunk 1 above)
Comment (haiweizh) [reply cpp/src/cider/function/FunctionLookupEngine.cpp]: Done.

File: cpp/src/cider/function/FunctionLookupEngine.cpp (hunk 2 above)
Comment (haiweizh) [reply cpp/src/cider/function/FunctionLookupEngine.cpp]: Done.

**Review by chaojun-zhang — APPROVED** [review]

File: cpp/src/cider/function/FunctionLookupEngine.cpp [hunk 3]
```diff
@@ -296,11 +298,9 @@ const FunctionDescriptor FunctionLookupEngine::lookupFunction(
             from_platform,
             from_platform_));
   }
-  const std::string& function_name = function_signature.func_name;
-  auto real_function_name = getRealFunctionName(function_name);
-  FunctionSignature function_signature_result = function_signature;
-  function_signature_result.func_name = real_function_name;
-  function_descriptor.func_sig = function_signature_result;
+  function_signature.func_name = getRealFunctionName(function_signature.func_name);
+  ;
```
Comment (yma11) [inline cpp/src/cider/function/FunctionLookupEngine.cpp:302]: Remove it.

**Review by yma11 — APPROVED** [review]

**Pull request merged**